src/DataLib/fendl32B_retrofit/file_handling.py

```diff
@@ -0,0 +1,70 @@
+# Import packages
+from pathlib import Path
+
+def get_isotope(stem):
+    """
+    Extract the element name and mass number from a given filename.
+
+    Arguments:
+        stem (str): Stem of a an ENDF (TENDL) and/or PENDF file, formatted
+            as f'{element}{mass_number}.ext'
+    Returns:
+        element (str): Chemical symbol of the isotope whose data is contained
+            in the file.
+        A (str): Mass number of the isotope, potentially including the letter
+            "m" at the end if the isotope is in a metastable state.
+    """
+
+    isomer_id = ''
+
+    upper_case_letters = [char for char in stem if char.isupper()]
+    lower_case_letters = [char for char in stem if char.islower()]
+    numbers = [str(char) for char in stem if char.isdigit()]
+
+    if len(lower_case_letters) == 0:
+        lower_case_letters = ['']
+    elif len(lower_case_letters) > 1:
+        isomer_id = lower_case_letters[-1]
+        lower_case_letters = lower_case_letters[:-1]
+    
+    element = upper_case_letters[0] + lower_case_letters[0]
+    A = ''.join(numbers) + isomer_id
+
+    return element, A
```

Comment:
It seems like there must be a simpler way to do this?  The files are always named `n-Z[z]AAA[m]`, right, where 
* Z is an uppercase letter
* [z] is an optional lower case letter
* A is a digit
* [m] is an optional ower case letter 'm'

so why not something like:
```
Z_start = 2
Z_end = Z_start + 1
if !stem[Z_end].isdigit():
    Z_end += 1
element = stem[Z_start:Z_end]

A_start = Z_end
A_end = A_start + 3
if !stem[-1].isdigit():
   A_end += 1
A = stem[A_start:A_end]
```

Reply:
regular expressions can make this even simpler in terms of number of lines, although maybe with a higher cognitive burden